operator/controllers/seldondeployment_prepackaged_servers.go

```diff
@@ -142,6 +143,93 @@ func (pi *PrePackedInitialiser) addTFServerContainer(mlDep *machinelearningv1.Se
 	return nil
 }
 
+func (pi *PrePackedInitialiser) addTritonServer(pu *machinelearningv1.PredictiveUnit, deploy *appsv1.Deployment, serverConfig *machinelearningv1.PredictorServerConfig) error {
+
+	c := utils.GetContainerForDeployment(deploy, pu.Name)
+	existing := c != nil
+
+	httpPort := int32(constants.TritonDefaultHttpPort)
+	grpcPort := int32(constants.TritonDefaultGrpcPort)
+	if pu.Endpoint.Type == machinelearningv1.GRPC {
+		grpcPort = pu.Endpoint.ServicePort
+	} else {
+		httpPort = pu.Endpoint.ServicePort
+	}
+	tritonUser := int64(1000)
+
+	cServer := &v1.Container{
+		Name: pu.Name,
+		Args: []string{
+			"/opt/tritonserver/bin/tritonserver",
+			"--grpc-port=" + strconv.Itoa(int(grpcPort)),
+			"--http-port=" + strconv.Itoa(int(httpPort)),
+			"--model-repository=" + DefaultModelLocalMountPath,
+		},
+		Ports: []v1.ContainerPort{
+			{
+				Name:          "grpc",
+				ContainerPort: grpcPort,
+				Protocol:      v1.ProtocolTCP,
+			},
+			{
+				Name:          "http",
+				ContainerPort: httpPort,
+				Protocol:      v1.ProtocolTCP,
+			},
+		},
+		ReadinessProbe: &v1.Probe{
+			Handler: v1.Handler{HTTPGet: &v1.HTTPGetAction{
+				Path: constants.TritonProbeReadyPath,
+				Port: intstr.FromString("http"),
+			}},
+			InitialDelaySeconds: 20,
+			TimeoutSeconds:      1,
+			PeriodSeconds:       10,
+			SuccessThreshold:    1,
+			FailureThreshold:    3,
+		},
+		LivenessProbe: &v1.Probe{
+			Handler: v1.Handler{HTTPGet: &v1.HTTPGetAction{
+				Path: constants.TritonProbeLivePath,
+				Port: intstr.FromString("http"),
+			}},
+			InitialDelaySeconds: 60,
+			TimeoutSeconds:      1,
+			PeriodSeconds:       10,
+			SuccessThreshold:    1,
+			FailureThreshold:    3,
+		},
+		SecurityContext: &v1.SecurityContext{
+			RunAsUser: &tritonUser,
+		},
+		VolumeMounts: []v1.VolumeMount{
+			{
+				Name:      machinelearningv1.PODINFO_VOLUME_NAME,
+				MountPath: machinelearningv1.PODINFO_VOLUME_PATH,
+			},
+		},
+	}
+	machinelearningv1.SetImageNameForPrepackContainer(pu, cServer, serverConfig)
+
+	if existing {
+		cServer.DeepCopyInto(c)
+	} else {
+		if len(deploy.Spec.Template.Spec.Containers) > 0 {
+			deploy.Spec.Template.Spec.Containers = append(deploy.Spec.Template.Spec.Containers, *c)
+		} else {
+			deploy.Spec.Template.Spec.Containers = []v1.Container{*c}
+		}
+	}
```

Comment:
Shouldn't this add `cServer` to the list of containers instead of `c`?

Reply:
Its copying the spec into an existing container definition if provided by the user.